src/index.js

```diff
@@ -1,56 +1,147 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 
-const parseAttribute = (value) => {
-  try {
-    // try to parse attribute values as JSON, e.g.
-    // - "1" -> 1
-    // - "true" -> true
-    // - "null" -> null
-    // - "[1, null, \"foo\"]" -> [1, null, 'foo']
-    return JSON.parse(value);
-  } catch (err) {
-    // or just return the raw value as string
-    return value;
-  }
+const bool = 'bool';
+const number = 'number';
+const string = 'string';
+const json = 'json';
+const event = 'event';
+
+const Types = { bool, number, string, json, event };
+
+const mapAttributeToProp = (node, name) => node[name];
+
+const mapEventToProp = (node, name) => {
+  // accessing properties instead of attributes here
+  // (autom. attribute parsing)
+  const value = node[name];
+
+  return (...origArgs) => {
+    // dispatch DOM event
+    const domEvent = new Event(name, { bubbles: true });
+    domEvent.origArgs = origArgs; // store original arguments from handler
```

Comment:
That's not a standard, I just added this in case someone wants to access the data submitted by the React handler. 

But after some research I found that the correct way to implement this is to use a `CustomEvent` with the `detail` property (needs a ponyfill for IE11)